config/config.go

```diff
@@ -996,3 +1002,11 @@ func (c *Config) parseHistogramBuckets() ([]float64, error) {
 	sort.Float64s(result)
 	return result, nil
 }
+
+func (c *Config) envOverrides() {
+	// Redis password
+	swarmRedisPassword := os.Getenv("SWARM_REDIS_PASSWORD")
```

Comment:
I think the idea https://github.com/zalando/skipper/issues/1702#issuecomment-769816602 was to have it the other way around: if flag is not set use what is in env variable, i.e. explicit flag overrides environment.

My personal opinion is that we don't need the flag at all. Having it leaves a chance for some users to use it in order to later discover that it leaked and have to rotate passwords and asses the damage.

Different tools use different approaches, e.g. `redis-cli` allows flag while `psql` does not.

But maybe I just do not see the usecase for the flag.

Reply:
> I think the idea [#1702 (comment)](https://github.com/zalando/skipper/issues/1702#issuecomment-769816602) was to have it the other way around: if flag is not set use what is in env variable, i.e. explicit flag overrides environment.

You are right. I red it a bit hasty and got it the other way around.

> Different tools use different approaches, e.g. `redis-cli` allows flag while `psql` does not.

Usually `cli` clients have two options for password. Pass it interactively or as an option (I see [psql](https://www.postgresql.org/docs/current/app-psql.html) does the same), but never as an environment variable.

On the other hand, servers usually set password from a configuration file with some exceptions that also use `cli` flags like `redis-server --requirepass 1234` but the specific one seems that hides it from `ps`

So, if you ask me, since `skipper` is a server and not a client, I would go with the second approach. The issue with `skipper` though is that its configuration options are also `cli` flags.

Bottom line is that I would prefer to have it  **ONLY** in configuration file and not as flag nor environment variable.
That gives you the ability to run multiple instances of your server (I cannot find a use case for that) in the same environment with different passwords.